libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/checkConditionalFragment.kt

```diff
@@ -21,6 +21,7 @@ internal fun checkConditionalFragments(definitions: List<GQLDefinition>): List<I
     when (it) {
       is GQLOperationDefinition -> scope.checkConditionalFragments(it.selectionSet.selections)
       is GQLFragmentDefinition -> scope.checkConditionalFragments(it.selectionSet.selections)
+      else -> {}
```

Comment:
Interesting. Is that triggering an error now or is it a warning we've had forever?

Reply:
Not sure why but now if I remove this now it compiles fine - I'm pretty sure it didn't while I was working on this PR 🤨 There is a warning in the IDE only though.

Reply:
The plot thickens... 🕵️ Not a big deal, was just curious. Keeping the `else` branch is better 👍